migrate.py

```diff
@@ -15,11 +26,54 @@ def main():
                 auth.set_role_type(role_type, result[role_type])
 
     programs, status_code = auth.list_programs()
-    if status_code == 200:
-        for program_id in programs:
-            program, status_code = auth.get_program(program_id)
-            if status_code == 200:
-                auth.add_program(program)
+    print(programs)
+    if status_code != 200:
+        print(f"Couldn't list programs: {programs} {status_code}")
+        sys.exit(1)
+
+    # start a dictionary of programs that we can then add dacs to
+    programs_dict = {}
+    for program_id in programs:
+        program, status_code = auth.get_program(program_id)
+        if status_code == 200:
+            programs_dict[program_id] = program
+            programs_dict[program_id]["dac_authorizations"] = {}
+
+    # need to call Vault to get the list of users
+    headers = {
+        "X-Vault-Token": authx.auth.get_vault_token_for_service()
+    }
+    url = f"{VAULT_URL}/v1/opa/users"
+    response = requests.request("LIST", url, headers=headers)
+    if response.status_code != 200:
+        print(f"Couldn't list users: {response.text} {response.status_code}")
+        sys.exit(1)
+
+    result = response.json()["data"]["keys"]
+    for user_id in result:
+        user_dict, status_code = auth.get_user(user_id)
+        if status_code != 200:
+            print(f"Couldn't get user {user_id}: {user_dict} {status_code}")
+            sys.exit(1)
```

Comment:
Same as below: if there's a problem with only specific users, it might make sense to just continue to the next loop iteration instead and just have the `sys.exit(1)` call at the end

Reply:
fixed now